### PR TITLE
V0.17.0

### PR DESCRIPTION
**Pull request by nmeyerhans**
Addresses #100 by updating the SDK for compatibility with Firecracker 0.17's API.

Certain attributes are now passed as pointers.

`GetMachineConfig` and related objects have been renamed to `GetMachineConfiguration` for consistency with `PutMachineConfiguration`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nmeyerhans** [comment]
Tests will fail until firecracker 0.17.0 installed on the CI hosts.

**Comment by xibz** [comment]
`version.go` file needs to be updated along with the `CHANGELOG.md`'s `Unreleased` section to reflect the correct version.

**Comment by nmeyerhans** [comment]
> `version.go` file needs to be updated along with the `CHANGELOG.md`'s `Unreleased` section to reflect the correct version.

Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * Fixes bug where context was not being used at all during startVM (#86)
 * Updates the jailer's socket path to point to the unix socket in the jailer's workspace (#86)
 * Fixes bug where default socketpath would always be used when not using jailer (#84).
+* Update for compatibility with Firecracker 0.17.x
```
Comment (xibz) [inline CHANGELOG.md]: The `# [ Unreleased ]` needs to be renamed to `# v0.17.0`

Comment (nmeyerhans) [reply CHANGELOG.md]: Not until we actually release a 0.17.0 version of the SDK, which is not what we're doing here.

**Comment by nmeyerhans** [comment]
A number of fields in the `models.MachineConfiguration` structure generated by OpenAPI have changed from direct references to pointers. As a result, the zero-value of these fields has changed from something vaguely meaningful (or at least technically valid) to a null pointer. This will require changes to client code that is relying on the default values for things like `VcpuCount`, `MemSizeMib`, and `HtEnabled`. I'm updating the validation code and tests to better handle the current behavior...

**Comment by nmeyerhans** [comment]
> Don't need the `nil` check since `Int64Value` will return `0` if the value is `nil`. Unless you did this intentionally for readability?

Yeah, it's kind of for readability. It's the nil value that we really care about. If the value isn't specified, then what we end up sending to firecracker isn't valid, and the API call to `PUT /machine-config` fails with `Missing mandatory fields.` The check for the validity of the numerical value is important, but a secondary consideration. (But maybe that's just a kind of recency bias on my part. I'm not opposed to changing the code if you care more than I do about it.)

**Comment by xibz** [comment]
> I'm not opposed to changing the code if you care more than I do about it.

Im okay with it. Just wanted to make sure the changes were intentional and not ignorant that `Int64Value` returns 0 when nil.

**Comment by nmeyerhans** [comment]
So, I've added code to fail `MachineConfiguration` validation if `VcpuCount`, `MemSizeMib`, or `HtEnabled` are unspecified. Another option would be to apply default values in that case, and proceed with validation. I'm inclined to leave the strict validation in place, since I don't really think it's a good idea for client code to be configuring VMs without making explicit choices about these settings. Does anybody disagree with me? @samuelkarp @sipsma @mxpv @xibz ?

**Comment by xibz** [comment]
@nmeyerhans - Yea, stricter validation and explicit choices is +1 for me.

:shipit:

**Comment by samuelkarp** [comment]
:+1: for strict validation. If we later decide to remove validation, we can do so.

File: machine.go [hunk 2]
```diff
@@ -126,6 +126,17 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("socket %s already exists", cfg.SocketPath)
 	}
 
+	if cfg.MachineCfg.VcpuCount == nil ||
```
Comment (samuelkarp) [inline machine.go]: I'd replace these conditionals with `assert.GreaterOrEqual(t, cfg.MachineCfg.VcpuCount, 1, "machine needs a nonzero VcpuCount")` and `assert.NotNil(t, cfg.MachineCfg.HtEnabled, "machine needs a setting for ht_enabled")`.

Comment (nmeyerhans) [reply machine.go]: This is in the actually validation code, not in the test code. So the right behavior is to return a validation failure error to the calling code.

Comment (samuelkarp) [reply machine.go]: Thanks, you are correct.

**Pull request merged**